awscrt/auth.py

```diff
@@ -255,6 +255,32 @@ def new_chain(cls, providers):
         binding = _awscrt.credentials_provider_new_chain(providers)
         return cls(binding)
 
+    @classmethod
+    def new_python(cls, py_provider):
```

Comment:
Any preference about this taking an object or a function instead? To make it simple, it may be better to be a simple function? @graebm

Reply:
I like this taking a function, that's much more common in python-land